apps/prairielearn/elements/pl-integer-input/pl-integer-input.py

```diff
@@ -159,17 +159,25 @@ def render(element_html: str, data: pl.QuestionData) -> str:
             # If answer is a string, convert back to an integer.
             # Else if answer is in a format generated by pl.to_json, convert it
             # back to a standard type. Otherwise do nothing.
-            if isinstance(a_sub, str):
-                a_sub_parsed = int(a_sub)
-            else:
-                a_sub_parsed = pl.from_json(a_sub)
+            try:
+                if isinstance(a_sub, str):
+                    # The representation of the submitted_answer is always
+                    # stored in base 10 by the parse function
+                    a_sub_parsed = int(a_sub)
+                else:
+                    a_sub_parsed = pl.from_json(a_sub)
+                a_sub = (
+                    numpy.base_repr(a_sub_parsed, base)
+                    if base > 0
+                    else data["raw_submitted_answers"].get(name, str(a_sub_parsed))
+                )
+            except ValueError:
+                # If the submitted answer is not a valid integer, use the raw
+                # submitted answer
+                a_sub = data["raw_submitted_answers"].get(name, a_sub)
```

Comment:
This handling of an exception is not strictly related to the main issue, but it can avoid the creation of an issue if the element type is changed after submissions are in place, as recently seen on Slack. As it were, the issue was cryptic and unrelated to the problem.